libfreerdp/core/gateway/http.c

```diff
@@ -1041,3 +1099,54 @@ TRANSFER_ENCODING http_response_get_transfer_encoding(HttpResponse* response)
 
 	return response->TransferEncoding;
 }
+
+BOOL http_response_is_websocket(HttpContext* http, HttpResponse* response)
+{
+	if (!http || !response)
+		return FALSE;
+	if (http->websocketUpgrade == TRUE && response->StatusCode == HTTP_STATUS_SWITCH_PROTOCOLS &&
+	    (response->SecWebsocketVersion == NULL ||
+	     _stricmp(response->SecWebsocketVersion, "13") == 0) &&
+	    response->SecWebsocketAccept)
+	{
+		// now check if Sec-Websocket-Accept is correct
+		WINPR_DIGEST_CTX* sha1 = NULL;
+		char* base64accept = NULL;
+		BYTE sha1_digest[WINPR_SHA1_DIGEST_LENGTH];
+		sha1 = winpr_Digest_New();
+		if (!sha1)
+			return FALSE;
+
+		if (!winpr_Digest_Init(sha1, WINPR_MD_SHA1))
+			goto out;
+
+		if (!winpr_Digest_Update(sha1, (const BYTE*)http->SecWebsocketKey,
+		                         strlen(http->SecWebsocketKey)))
+			goto out;
+		if (!winpr_Digest_Update(sha1, (const BYTE*)"258EAFA5-E914-47DA-95CA-C5AB0DC85B11",
+		                         strlen("258EAFA5-E914-47DA-95CA-C5AB0DC85B11")))
+			goto out;
+
+		if (!winpr_Digest_Final(sha1, sha1_digest, sizeof(sha1_digest)))
+			goto out;
+
+		base64accept = crypto_base64_encode(sha1_digest, WINPR_SHA1_DIGEST_LENGTH);
+		if (!base64accept)
+			goto out;
+
+		if (_stricmp(response->SecWebsocketAccept, base64accept) != 0)
+		{
+			WLog_WARN(TAG, "Webserver gave Websocket Upgrade response but sanity check failed");
+			goto out;
+		}
+		winpr_Digest_Free(sha1);
```

Comment:
Duplicated code, use a return variable defaulting to `FALSE` set to `TRUE` here instead